var/spack/repos/builtin/packages/tree/package.py

```diff
@@ -19,13 +19,14 @@ class Tree(Package):
     homepage = "http://mama.indstate.edu/users/ice/tree/"
     url = "http://mama.indstate.edu/users/ice/tree/src/tree-1.7.0.tgz"
 
+    version("2.0.3", sha256="ba14e77b5f9dc7f8250c3f702ec5b6be2f93cd0fa87311bab3239676866a3b1d")
     version("2.0.2", sha256="7d693a1d88d3c4e70a73e03b8dbbdc12c2945d482647494f2f5bd83a479eeeaf")
     version("1.8.0", sha256="715d5d4b434321ce74706d0dd067505bb60c5ea83b5f0b3655dae40aa6f9b7c2")
     version("1.7.0", sha256="6957c20e82561ac4231638996e74f4cfa4e6faabc5a2f511f0b4e3940e8f7b12")
 
     @when("@2:")
     def install(self, spec, prefix):
-        make("install", "PREFIX=%s" % prefix)
+        make("install", "PREFIX=%s" % prefix, "CFLAGS=-std=c99")
```

Comment:
Let's just stick to the hacky business for now...

```suggestion
        make(
            "PREFIX=%s" % prefix,
            "CC=%s" % spack_cc,
            "CFLAGS=-O3 -std=c99 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64",
            "install",
        )
```

Or if you really don't like copying the  defines, you can also inject `-O3 -std=c99` through the compiler wrapper:

```python
    def flag_handler(self, name, flags):
        if name == 'cflags':
            flags.append("-O3")
            flags.append("-std=c99")
        return (flags, None, None)
```

It would still be worth it passing `"CC=%s" % spack_cc,` to `make(...)` since they set `CC` in the makefile too.

Reply:
I'd prefer it the hacky way because the flags can be set properly, given the problems I noted in <https://github.com/spack/spack/pull/32463#issuecomment-1233406327>. I'll push an updated version shortly.

Reply:
I am also adding the commented-out `LDFLAGS`.

Does the order of the variables and `install` matter? I noticed you swapped them.